src/test/java/com/github/dockerjava/core/command/WaitContainerCmdImplTest.java

```diff
@@ -104,4 +102,31 @@ public void testWaitContainerAbort() throws Exception {
 
         assertThat(inspectContainerResponse.getState().getRunning(), is(equalTo(false)));
     }
+
+    @Test
+    public void testWaitContainerTimeout() throws Exception {
+
+        CreateContainerResponse container = dockerClient.createContainerCmd("busybox").withCmd("sleep", "9999").exec();
+
+        LOG.info("Created container: {}", container.toString());
+        assertThat(container.getId(), not(isEmptyString()));
+
+        dockerClient.startContainerCmd(container.getId()).exec();
+
+        WaitContainerResultCallback callback = dockerClient.waitContainerCmd(container.getId()).exec(
+                new WaitContainerResultCallback());
+        try {
+            callback.awaitStatusCode(100, TimeUnit.MILLISECONDS);
+            fail("Should throw exception on timeout.");
+        } catch(DockerClientException e){
+            LOG.info(e.getMessage());
+        }
+
+        dockerClient.killContainerCmd(container.getId()).exec();
```

Comment:
Why you killing container here? Container created while test will be removed automatically after test.